src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityAssemblyLine.java

```diff
@@ -60,4 +70,119 @@ public ICubeRenderer getBaseTexture(IMultiblockPart sourcePart) {
     protected IBlockState getCasingState() {
         return MetaBlocks.METAL_CASING.getState(BlockMetalCasing.MetalCasingType.STEEL_SOLID);
     }
+
+    private int beamCount;
+    @SideOnly(Side.CLIENT)
+    private GTLaserBeamParticle[][] beamParticles;
+
+    @Override
+    public void update() {
+        super.update();
+        if(ConfigHolder.client.shader.assemblyLineParticles) {
+            if (getRecipeMapWorkable().isWorking()) {
+                int count = (int) (getRecipeMapWorkable().getProgressPercent() * getAbilities(MultiblockAbility.IMPORT_ITEMS).size()) + 1;
+                if (count != beamCount) {
+                    beamCount = count;
+                    writeCustomData(GregtechDataCodes.UPDATE_PARTICLE, this::writeParticles);
+                }
+                return;
+            }
+            if (beamCount != 0) {
```

Comment:
Can we keep the lasers emitting when the machine is paused?
Pausing multiblock does not mean resetting its progress so it is a more natural behaviour
(Also, if you can, I want to see beams count are decreasing when Assline has not enough energy)